### PR TITLE
Home task01

### PR DESCRIPTION
**Pull request by vnormatov**

### CODE REVIEW

**Comment by pavlom0x** [comment]
Description very general
you should describe in pull request what functionality you added
mention what it should solve/do

### Example

Added methods for solving tasks
- watermelon
- nextYearFind
- tests for methods

File: entry-test/beautifulyear.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+'use strick';
```
Comment (pavlom0x) [inline entry-test/beautifulyear.js]: `'use strict'` check all files

**Review by pavlom0x — CHANGES_REQUESTED** [review]

File: entry-test/chat.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+'use strick';
+
+function isSucceessSayHello(phrase) {
```
Comment (yuchiko) [inline entry-test/chat.js]: typo in function name `isSucceessSayHello` should be `isSuccessSayHello`

File: entry-test/beautifulyear.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+function getYear(year) {
+    for (var i = year; i < 9013; i++) {
+        if (check(i) == true) {
+            return i;
```
Comment (yuchiko) [inline entry-test/beautifulyear.js]: bad naming, it will be good if you will use something like `nextYear` not `i`

File: entry-test/chat.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    let phraseArray = phrase.split('');
+    let result = '';
+    for (let obj of phraseArray) {
```
Comment (yuchiko) [inline entry-test/chat.js]: `obj` ?

File: entry-test/chat.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        switch (obj) {
+            case 'h':
+                if (result.length < 1)
+                    result += 'h';
+                break;
+            case 'e':
+                if (!result.localeCompare('h'))
```
Comment (yuchiko) [inline entry-test/chat.js]: `localeCompare` is redundant here, it's enough to use equal compare
`result === 'h'` same to other cases

File: entry-test/beautifulyear.js (hunk 1 above)
Comment (vnormatov) [reply entry-test/beautifulyear.js]: ok))

File: entry-test/watermelon.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+'use strick';
+
+function makeDecision(weight) {
+    if (weight < 4 || weight % 2 > 0) {
+        return "NO"
```
Comment (pavlom0x) [inline entry-test/watermelon.js]: better to use single quote `'`

File: entry-test/watermelon.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    }
+    else return 'YES';
```
Comment (pavlom0x) [inline entry-test/watermelon.js]: don't mix `'` and `"`

File: entry-test/watermelon.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+};
+
+function makeDecisionType(weight) {
```
Comment (pavlom0x) [inline entry-test/watermelon.js:10]: what's a difference with `makeDecision` ?

File: entry-test/stones.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
+'use strick';
+
+function getMinStonesToDell(numberOfStones, line) {
```
Comment (pavlom0x) [inline entry-test/stones.js]: `numberOfStones` not used
do you actually need it?

File: test/stones-spec.js [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+        quantity.should.be.eql(3);
+    });
+
+    it('count quantity of stones for deleting if line size is 4 and  stones follow as BRBG ', function () {
+        let quantity = algolib.stones.getMinStonesToDell(4, 'BRBG');
+        quantity.should.be.eql(0);
+    });
+
+    it('count quantity of stones for deleting if line size is 2 and  stones follow as BG ', function () {
+        let quantity = algolib.stones.getMinStonesToDell(2, 'BRBG');
```
Comment (pavlom0x) [inline test/stones-spec.js]: description collision

File: test/watermelon-spec.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+        let decision = algolib.watermelon.makeDecision(1);
+        decision.should.be.eql("NO");
+    });
+
+    it('make decision if weight=150', function () {
+        let decision = algolib.watermelon.makeDecision(150);
+        decision.should.be.eql("YES");
+    });
+
+    it('make decision if weight is even', function () {
```
Comment (pavlom0x) [inline test/watermelon-spec.js]: How is this case different with previous one?

**Review by pavlom0x — CHANGES_REQUESTED** [review]

**Review by yuchiko — CHANGES_REQUESTED** [review]

File: entry-test/watermelon.js (hunk 8 above)
Comment (vnormatov) [reply entry-test/watermelon.js:10]: I did it only for me, to understand how "module.exports" works

File: test/watermelon-spec.js (hunk 11 above)
Comment (vnormatov) [reply test/watermelon-spec.js]: I checked range limits of input values, odd and even numbers. Previous is a limit of input range.

File: entry-test/watermelon.js (hunk 7 above)
Comment (vnormatov) [reply entry-test/watermelon.js]: fixed

File: test/stones-spec.js (hunk 10 above)
Comment (vnormatov) [reply test/stones-spec.js]: fixed

File: entry-test/chat.js (hunk 4 above)
Comment (vnormatov) [reply entry-test/chat.js]: fixed

File: entry-test/chat.js (hunk 5 above)
Comment (vnormatov) [reply entry-test/chat.js]: fixed

File: entry-test/beautifulyear.js (hunk 3 above)
Comment (vnormatov) [reply entry-test/beautifulyear.js]: fixed

**Review by pavlom0x — APPROVED** [review]

**Review by yuchiko — APPROVED** [review]

**Pull request merged**